src/psyclone/psyir/backend/sir.py

```diff
@@ -403,10 +411,15 @@ def literal_node(self, node):
 
         '''
         result = node.value
-        # There is an assumption here that the literal is a float (see
-        # #612).
-        return ("{0}make_literal_access_expr(\"{1}\", BuiltinType.Float)"
-                "".format(self._nindent, result))
+        try:
+            datatype = TYPE_MAP_TO_SIR[node.datatype]
+        except KeyError:
+            raise VisitorError(
+                "PSyIR type '{0}' has no representation in the SIR backend."
+                "".format(str(node.datatype)))
+        
```

Comment:
Trailing whitespace.

Reply:
And again. There was also a complaint about using not in the wrong place which I've also changed. pycodestyle and pylint are now happy.